includes/Core/Modules/Module_Sharing_Settings.php

```diff
@@ -127,28 +127,24 @@ public function get() {
 	/**
 	 * Merges a partial Module_Sharing_Settings option array into existing sharing settings.
 	 *
-	 * Only updates sharing settings for a module if the current user has the capability to
-	 * do so.
-	 *
 	 * @since 1.75.0
+	 * @since n.e.x.t Removed capability checks.
 	 *
-	 * @param array $new_partial_settings Partial settings array to update existing settings with.
+	 * @param array $partial Partial settings array to update existing settings with.
 	 *
 	 * @return bool True if sharing settings option was updated, false otherwise.
 	 */
-	public function merge( array $new_partial_settings ) {
+	public function merge( array $partial ) {
 		$settings = $this->get();
-
-		foreach ( $new_partial_settings as $module_slug => $new_settings ) {
-			if ( null === $new_settings ) {
-				continue;
+		$partial  = array_filter(
+			$partial,
+			function ( $value ) {
+				return null !== $value;
 			}
-			if ( current_user_can( Permissions::MANAGE_MODULE_SHARING_OPTIONS, $module_slug ) ) {
-				$settings[ $module_slug ] = $new_settings;
-			}
-		}
+		);
+		$updated  = array_intersect_key( $partial, $settings );
 
-		return $this->set( $settings );
+		return $this->set( array_merge( $settings, $updated ) );
```

Comment:
We can't just merge the original settings with updated ones. The problem here is that updated settings may not contain certain subkeys if the user doesn't have some of required capabilities which will lead to a removal of missing subkeys in the updated settings from the original settings. 

For example, let's assume we get the following updated settings:

```php
$updated = array(
    'moduleA' => array(
        'management' => 'owner',
    )
)
```
As you can see there, the `moduleA` lacks the `sharedRoles` property which has been unset in the `Collection_Key_Cap_Filter` class because the user doesn't have the `MANAGE_MODULE_SHARING_OPTIONS` capability.

Now if we have the following settings, then `sharedRoles` will be removed after merging both arrays.
```php
$settings = array(
    'moduleA' => array(
        'management' => 'all_admins',
        'sharedRoles' => array( 'role1', 'role2' ),
    )
)
```
We need to preserve the current value of the `sharedRoles` property for the `moduleA`.